scripts/prompt-config.sh

```diff
@@ -235,11 +235,17 @@ is_email() {
 	return 1
 }
 
+#
+# Validate a DNS-style value as per
+# http://en.wikipedia.org/wiki/Domain_Name_System#Domain_name_syntax
+#
+# Max 63 chars, alphanumeric and hyphen, can't start or end with hyphen,
+# can't be *all* numeric. We additionally enforce all lower-case: the
+# region is sometimes used in case-sensitive scenarios like Zookeeper
+# nodes.
+#
 is_dns_label() {
-	# http://en.wikipedia.org/wiki/Domain_Name_System#Domain_name_syntax
-	# Max 63 chars, alphanumeric and hypen, can't start or end with hyphen,
-	# can't be *all* numeric.
-	if [[ "$1" =~ ^[a-zA-Z0-9][a-zA-Z0-9-]*$ ]] \
+	if [[ "$1"  =~ ^[[:lower:]0-9][[:lower:]0-9-]*$ ]] \
```

Comment:
curious why bother with `[:lower:]` instead of `a-z`?

Reply:
See my scrum yesterday! Under the default locale, a-z collation range includes the upper-case chars. Sad but true.